src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -454,6 +457,10 @@ public Response get(@Context HttpHeaders httpHeaders, String json) throws Except
 
 			for (QueryMetric query : queries)
 			{
+				if (System.currentTimeMillis()-start > datastore.getM_queryTimeout()){
```

Comment:
If an individual query that started before the timeout hangs, Kairos will continue to hang until that query completes no matter how long it takes.

Let's look into the C* client itself to see if we can do something there instead.

Reply:
Possibly this, for example: https://docs.datastax.com/en/drivers/java/3.5/com/datastax/driver/core/SocketOptions.html#setReadTimeoutMillis-int-

Reply:
We can also set this server-side as well - it would be worth looking at what we're setting that to there, if anything at all, in which case then what the default values are.

Reply:
right.. makes sense. will set a query level timeout as well in combination with the response timeout

Reply:
@beverage timeouts for each query are set to at the next server level - cassandra. So we should be safe to say that the query will max hang till that server level timeout. If we set our response timeout appropriately as per the controller timeouts, we should be good. What do you say?